bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/StylesheetInlinerTransformerFactory.java

```diff
@@ -0,0 +1,200 @@
+        label = "Stylesheet Inliner Transformer Factory", 
+        description = "Sling Rewriter Transformer Factory which inlines CSS references")
+@Properties({ 
+    @Property( name = "pipeline.type", value = StylesheetInlinerTransformerFactory.SELECTOR, propertyPrivate = true) } )
+@Service(value = { TransformerFactory.class })
+public final class StylesheetInlinerTransformerFactory implements TransformerFactory {
+
+    public static final String SELECTOR = "inline-css";
+    
+    private static final String NEWLINE = "\n";
+    private static final String STYLE = "style";
+    private static final String BODY = "body";
+    private static final String HEAD = "head";
+
+    private static final Logger log = LoggerFactory.getLogger(StylesheetInlinerTransformerFactory.class);
+
+    @Reference
+    private HtmlLibraryManager htmlLibraryManager;
+
+
+    public Transformer createTransformer() {
+        return new CSSInlinerTransformer();
+    }
+
+    private class CSSInlinerTransformer extends AbstractTransformer {
+
+        protected boolean enabled = false;
+        protected boolean inHead = false;
+        protected List<String> stylesheetsInHead = new ArrayList<String>();
+        private SlingHttpServletRequest slingRequest;
+
+        @Override
+        public void init(ProcessingContext context, ProcessingComponentConfiguration config) throws IOException {
+            super.init(context, config);
+            slingRequest = context.getRequest();
+
+            String[] selectors = slingRequest.getRequestPathInfo().getSelectors();
+
+            if (selectors.length > 0) {
+                enabled = Arrays.asList(selectors).contains(SELECTOR);
+                if (enabled) {
+                    log.info("Inlining Stylesheet references for {}", slingRequest.getRequestURL().toString());
+                }
+            }
+        }
+
+        public void startElement(final String namespaceURI, final String localName, final String qName,
+                final Attributes attrs) throws SAXException {
+            if (enabled) {
+                try {
+                    if (isCSS(localName, attrs)) {
+                        String sheet = attrs.getValue("", "href");
+                        if (inHead) {
+                            stylesheetsInHead.add(sheet);
+                        } else {
+                            log.debug("Inlining stylesheet link found in BODY: '{}'", sheet);
+                            inlineSheet(namespaceURI, sheet);
+                        }
+                    } else {
+                        getContentHandler().startElement(namespaceURI, localName, qName, attrs);
+
+                        if (localName.equalsIgnoreCase(HEAD)) {
+                            inHead = true;
+                        } else if (localName.equalsIgnoreCase(BODY)) {
+
+                            // add each of the accumulated stylesheet references
+                            for (String sheet : stylesheetsInHead) {
```

Comment:
It'd simplify the code a bit to move it to </head>, and I don't see that there's any difference in support in the campaignmonitor document.  Moving.